[Bug]: intractable error: TypeError: only length-1 arrays can be converted to Python scalars /retrieval

### Is there an existing issue for the same bug?

- [X] I have checked the existing issues.

### Branch name

main

### Commit ID

2a614e0

### Other environment information

```Markdown
MacOS 15.0.1 Sequoia
M3 Max, 128 GB RAM, 2TB disk
```


### Actual behavior

Attempting to use /retrieval results in `TypeError: only length-1 arrays can be converted to Python scalars`. I am providing only `question`, `datasets`, and `documents`. I added debug logs and got the following stack trace below.


### Expected behavior

Expected results of query. Everything is showing as ingested successfully.

### Steps to reproduce

```Markdown
1. Install the docker container
2. Upload documents to one dataset
3. Start parse via API and wait to complete
4. Call /api/v1/retrieval with a question, that dataset id, and the list of document ids
5. Unknown error
```


### Additional information


```
2024-10-23 16:48:57 [ERROR] [2024-10-23 16:48:57,071] [doc.retrieval_test] [line:651]: Error during retrieval: only length-1 arrays can be converted to Python scalars
2024-10-23 16:48:57 Traceback (most recent call last):
2024-10-23 16:48:57   File "/ragflow/api/apps/sdk/doc.py", line 623, in retrieval_test
2024-10-23 16:48:57     ranks = retr.retrieval(question, embd_mdl, kb.tenant_id, kb_ids, page, size,
2024-10-23 16:48:57             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2024-10-23 16:48:57   File "/ragflow/rag/nlp/search.py", line 374, in retrieval
2024-10-23 16:48:57     sres = self.search(req, index_name(tenant_id), embd_mdl, highlight)
2024-10-23 16:48:57            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2024-10-23 16:48:57   File "/ragflow/rag/nlp/search.py", line 129, in search
2024-10-23 16:48:57     s["knn"] = self._vector(
2024-10-23 16:48:57                ^^^^^^^^^^^^^
2024-10-23 16:48:57   File "/ragflow/rag/nlp/search.py", line 64, in _vector
2024-10-23 16:48:57     "query_vector": [float(v) for v in qv]
2024-10-23 16:48:57                      ^^^^^^^^
2024-10-23 16:48:57 TypeError: only length-1 arrays can be converted to Python scalars
2024-10-23 16:48:57 [INFO] [2024-10-23 16:48:57,073] [_internal._log] [line:97]: 172.19.0.2 - - [23/Oct/2024 16:48:57] "POST /api/v1/retrieval HTTP/1.1" 200 -         
```

--- a/rag/nlp/search.py
+++ b/rag/nlp/search.py
@@ -46,6 +46,9 @@ class Dealer:
 
     def get_vector(self, txt, emb_mdl, topk=10, similarity=0.1):
         qv, _ = emb_mdl.encode_queries(txt)
+        shape = np.array(qv).shape
+        if len(shape) > 1:
+            raise Exception(f"Dealer.get_vector returned array's shape {shape} doesn't match expectation(exact one dimension).")
         embedding_data = [float(v) for v in qv]
         vector_column_name = f"q_{len(embedding_data)}_vec"
         return MatchDenseExpr(vector_column_name, embedding_data, 'float', 'cosine', topk, {"similarity": similarity})

